File: Arrays_and_Hashing/Complete_Search/Bronze/P2_2020-Daisy_Chains/DaisyChains.java
import java.io.*;
import java.util.*;

public class DaisyChains {
	public static void main(String[] args) throws IOException {
		Scanner sc = new Scanner(System.in);
		// Read in 1st line: n = # of flowers
		final int N = sc.nextInt();
		// Read in 2nd line: the petals of each flower
		int[] flowerPetals = new int[N];
		for (int i = 0; i < N; i++)
			flowerPetals[i] = sc.nextInt();
		sc.close();

		int avgFlowerPicCount = N;
		for (int i = 0; i < N - 1; i++) {
			int sum = flowerPetals[i];

			Map<Integer, Integer> petalCount = new HashMap<>();
			petalCount.put(flowerPetals[i], 1);
			for (int j = i + 1; j < N; j++) {
				sum += flowerPetals[j];

				petalCount.put(
					flowerPetals[j],
					petalCount.getOrDefault(flowerPetals[j], 0) + 1
				);

				int length = j - i + 1;
				if (sum % length != 0)
					continue;

				// Check # of flowers with the current average # of petals in the window range
				if (petalCount.getOrDefault(sum / length, 0) > 0)
					avgFlowerPicCount++;
			}
		}

		System.out.println(avgFlowerPicCount);
	}
}

File: Arrays_and_Hashing/Complete_Search/Bronze/P2_2020-Daisy_Chains/daisychains.py
# Read in 1st line: n = # of flowers
N = int(input())
# Read in 2nd line: the petals of each flower
flower_petals = list(map(int, input().split()))

avg_count = N
for i in range(N - 1):
    current_sum = flower_petals[i]

    petal_count = [0] * 1000
    petal_count[flower_petals[i]] += 1
    for j in range(i + 1, N):
        current_sum += flower_petals[j]
        petal_count[flower_petals[j]] += 1

        length = j - i + 1
        if current_sum % length != 0:
            continue

        if petal_count[current_sum // length] > 0:
            avg_count += 1

print(avg_count)
